docs: add readme files to packages (#208)
Azure-Samples/contoso-real-estate: packages/portal/README.md

It will also instruct you how to deploy it to [Azure Static Web Apps](https://learn.microsoft.com/azure/static-web-apps/overview), to publish it to the cloud, independenty, using the [Azure Static Web Apps CLI](https://azure.github.io/static-web-apps-cli/).

## Prerequisites

**IMPORTANT: THIS SCENARIO IS TIGHTLY COUPLED WITH SCENARIO 3. SOME PARTS OF THIS APP MAY NOT WORK AS EXPECTED IF YOU DON'T FOLLOW THE INSTRUCTIONS IN SCENARIO 3.**

In order to start the development server for local development or locally browsing the portal site, the following technologies must be installed in your computer:

- [node.js](https://nodejs.org) **v18.15.0** specifically, with the corresponding npm version
- [Azure Core Tools](https://learn.microsoft.com/azure/azure-functions/functions-run-local)
- [Azure Static Web Apps CLI](https://azure.github.io/static-web-apps-cli/)

### Angular CLI

This project was generated with [Angular CLI](https://github.com/angular/angular-cli) version 14.2.3.

## Steps to start the portal

1. fork or clone the repository locally

> 2. assumming you are in the folder containing your clone, go to the terminal and run 

Reviewer: ```suggestion
2. assumming you are in the folder containing `contoso-real-estate/packages/portal`, go to the terminal and run 
```

```bash

> contoso-real-estate && npm run clean:install

Reviewer: ```suggestion
npm run clean:install
```

```
    
at the root level of the respository. This will install all dependencies for all scenarios. 

This operation will also install the [Azure Static Web Apps CLI](https://azure.github.io/static-web-apps-cli/docs/intro). This tool includes a local dev server and emulator, to test the application and the corresponding API together, locally.

> **NOTE:** _If you want to work with this web app only, you can follow instructions to start an Angular application locally, by going to the folder `contoso-real-estate/packages/portal` and running_
> 
> ```bash
> npm install && npm start
> ```

Reviewer: ```suggestion
```

> 3. assuming you are now in the `contoso-real-estate` or _root_ folder, go to the terminal and run

Reviewer: ```suggestion
3. assuming you are now in the `contoso-real-estate`, go to the terminal and run
```

```bash
npm start
```

Follow the [Azure Static Web Apps CLI](https://azure.github.io/static-web-apps-cli/docs/cli/swa) prompts to complete the configuration.
    
Your default browser should open a new window with the application running. If you did not pass a port option, it should be running at [http://localhost:4280](http://localhost:4280)

> 4. Assuming you are in the folder `contoso-real-estate/packages/portal` and that you have started the Angular application in the default port 4200, start the emulator by executing 
> 
> ```bash
> swa start http://localhost:4200 --api-location ../api
> ```

Reviewer: ```suggestion
```

5. Go to `http://localhost:4280`. The application and the API should be running and functional. You should be able to see the homepage and the cards or teasers for the listings. You should be able to navigate to any of the listings detail page by clicking in any of the cards.

## Troubleshooting the local setup

- Q: The application server starts up but I can't see the listings
- A: Go to `http://localhost:7071/api/listings` and make sure you see a JSON printed in the browser.

> - Q: The application server starts up but I can't navigate to any of the listings detail pages
> - A: Go to `http://localhost:7071/api/listings` and make sure you see a JSON printed in the browser

Reviewer: ```suggestion
- Q: The application server starts up but I can't navigate to any of the listings detail pages
- A: Go to `http://localhost:7071/api/listings` and make sure you see a JSON printed in the browser

- Q: The application server starts up but I can't navigate to any of the listings detail pages
- A: Make sure you are accessing the portal served by the SWA CLI `http://localhost:4280` (and not the Angular CLI).
```
